models/model.go

```diff
@@ -478,7 +616,7 @@ func GetEntityHierarchy(ID primitive.ObjectID, ent string, start, end int) (map[
 			}
 		}
 
-		if children != nil && len(children) > 0 {
```

Comment:
I suggest to keep the checking for nil, to prevent crashing because we cannot check the length of a nil data type

Reply:
It's because of this: should omit nil check; len() for []map[string]interface{} is defined as zero (S1009)go-staticcheck